### DISCUSSION TITLE
12.3.0  register services error，thks

### DISCUSSION THREAD

**Discussion by dean1979** [LuckyPennySoftware/MediatR#1042] [Q&A]
RegisterServicesFromAssembly(Assembly.GetExecutingAssembly());

![image](https://github.com/jbogard/MediatR/assets/6007483/ff8cf3ac-c1d2-48f7-bda5-8e53f4526524)

**Comment by zachpainter77** [comment]
This issue has been tracked here: #1047 

And a PR has been created fixing this issue here: #1048

**Comment by zachpainter77** [comment]
The fix for this is to:

1. Rollback to the previous version of mediator
OR
2. Add configuration to tell MediatR to not try to register generic types.
```csharp
builder.Services.AddMediatR(config => {
    config.TypeEvaluator = x => !x.ContainsGenericParameters;
    config.RegisterServicesFromAssemblies(assemblies);
});
```

**Reply by zachpainter77** [reply]
This is a work around until the bug fix has been merged to master.
